diff --git a/cmd/caddy/main.go b/cmd/caddy/main.go
index 48fa149a..cf1c77f7 100644
--- a/cmd/caddy/main.go
+++ b/cmd/caddy/main.go
@@ -32,6 +32,7 @@ import (
 	caddycmd "github.com/caddyserver/caddy/v2/cmd"

 	// plug in Caddy modules here
+	_ "github.com/caddy-dns/cloudflare"
 	_ "github.com/caddyserver/caddy/v2/modules/standard"
 )

diff --git a/go.mod b/go.mod
index eaff3a5b..1c117074 100644
--- a/go.mod
+++ b/go.mod
@@ -7,6 +7,7 @@ require (
 	github.com/Masterminds/sprig/v3 v3.2.3
 	github.com/alecthomas/chroma/v2 v2.5.0
 	github.com/aryann/difflib v0.0.0-20210328193216-ff5ff6dc229b
+	github.com/caddy-dns/cloudflare v0.0.0-20230224210553-a9d3ae2690a1
 	github.com/caddyserver/certmagic v0.17.2
 	github.com/dustin/go-humanize v1.0.1
 	github.com/go-chi/chi v4.1.2+incompatible
@@ -50,6 +51,7 @@ require (
 	github.com/golang/mock v1.6.0 // indirect
 	github.com/google/pprof v0.0.0-20210407192527-94a9f03dee38 // indirect
 	github.com/grpc-ecosystem/grpc-gateway/v2 v2.7.0 // indirect
 	github.com/onsi/ginkgo/v2 v2.2.0 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/quic-go/qpack v0.4.0 // indirect
+	github.com/libdns/cloudflare v0.1.0 // indirect
